### DISCUSSION TITLE
3.21.23.0 Update Q and A

### DISCUSSION THREAD

**Discussion by Ongjming** [SmartlyDressedGames/Unturned-3.x-Community#2706] [Q&A] [answered]
1. Deadzone full suit protection option.
Does it mean that "Proof_Radiation" can be applied to all clothing? (Shirt, Pant, Mask, Hat? for a set?)
(Haven't tried it out so have no clue how it works in game yet. e.g. how it drains durability, what does it do, extend timer?)

2. Level asset skill default and max level overrides. (https://steamcommunity.com/app/304930/discussions/3/5183064363535019558/)
That guy was asking about it and I thought to look at https://github.com/SmartlyDressedGames/U3-Docs/blob/master/LevelAsset.md in the documents.
(Skills array: overrides skill default and max levels. Refer to skill rule properties.
and
Skill Rule Properties
Id string: Name of skill, for example Sharpshooter.
Default_Level int: Skill level when player spawns. Note server config Spawn_With_Max_Skills takes priority.
Max_Unlockable_Level int: Maximum skill level attainable through gameplay. Higher levels are hidden in the skills menu.)

I feel what he asked was still valid, and was hoping for more clarity on how this works as well.
I am currently assuming that with the settings, we can limit the max skill level to be _lower_ than the default vanilla's
and that there isn't actually anyway for us to increase the skill level to be higher than vanilla default for other modding purposes currently.

3. Trap_Setup_Delay and Trap_Cooldown options for traps.
I am assuming this goes into the trap item's .dat file as lines
and that Trap_Setup_Delay = Amount of time in ticks before the place down animation would occur?
Trap_Cooldown = Once someone triggered the trap and hurt them, it won't hurt anything for the duration of this, also in ticks?

Thanks if anyone who tried them out knew more about these and share here for posterity. :)

**Comment by iBowie** [comment, marked as answer]
1. It works only for shirt and pants and only in certain deadzone types, they are specified like 'FullSuitRadiation'
2. 'placeholder for github to actually handle numbered list correctly'
3. 'Trap_Setup_Delay' as from i've seen is how many seconds should pass before placed trap will be triggerable, and 'Trap_Cooldown' is how many seconds should pass from the moment it was last triggered to be active again.

upd: for the first one - it does not drain durability at all

**Reply by SDGNelson** [reply]
Thanks iBowie, yes these answers are correct.

2. Currently yes it only limits the level, cannot raise above the vanilla max level (yet). For example if the vanilla max level is 10 and max unlockable level is 5 then the skill is capped at 50% efficiency.

**Reply by Ongjming** [reply]
(I am glad that there are more information and more discussion coming out of this :) )
Thought to reply here again because after thinking about it, wanted to know if my subsequent assumption was correct:
FullSuitRadiation
As a line in the .dat file for Shirt and Pants type items, without true/false or anything else, when worn together in game would provide complete protection vs deadzone's radiation without a need for gasmask.
(Something like "Invulnerable" tag on gun type items without true/false and able to effect something)

**Reply by SDGNelson** [reply]
No, a gas mask is currently still necessary in both types of deadzone.

**Reply by Ongjming** [reply]
> both types of deadzone.

Sorry, to deviate a little, just want to ask this:
I am not aware that there are types of deadzones (My current knowledge for deadzone: Deadzone = Deadzone nodes)
May I know what types of deadzones are there currently?

**Reply by SDGNelson** [reply]
To clarify: full suit requirement is enabled on the new deadzone type. Previously there was no type and only required gasmasks. Shirts and pants with the Proof_Radiation key protect against the new type.

**Comment by WhiteBlaaack** [comment]
How should I config the skill asset?
I've tried to make it like this but obviously it's wrong
"Metadata"
{
	"GUID" "acde0f9413a443dfba9cc6d96334f4da"
	"Type" "SDG.Unturned.SkillsAsset, Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null"
}
"Asset"
{
	"ID" "Overkill"
	"Default_Level" "0"
	"Max_Unlockable_Level" "0"
}

**Reply by SDGNelson** [reply]
Close, you just need to change:
```
"ID" "Overkill"
"Default_Level" "0"
"Max_Unlockable_Level" "0"
```
to
```
"Skills"
[
    {
        "Id" "Overkill"
        "Default_Level" "0"
        "Max_Unlockable_Level" "0"
    }
]
```

**Reply by WhiteBlaaack** [reply]
![image](https://user-images.githubusercontent.com/74914249/126458850-e7e55204-e964-4b56-9f95-68147a0d60e2.png)
What does it mean?

**Reply by SDGNelson** [reply]
What's in your EcstasySkills.asset file now?

**Reply by WhiteBlaaack** [reply]
"Metadata"
{
	"GUID" "acde0f9413a443dfba9cc6d96334f4da"
	"Type" "SDG.Unturned.SkillsAsset, Assembly-CSharp, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null"
}
"Asset"
{
    "Skills"
    [
        {
        "Id" "Overkill"
        "Default_Level" "0"
        "Max_Unlockable_Level" "0"
        }
        {
        "Id" "Sharpshooter"
        "Default_Level" "0"
        "Max_Unlockable_Level" "0"
        }
        {
        "Id" "Dexterity"
        "Default_Level" "3"
        "Max_Unlockable_Level" "3"
        }
        {
        "Id" "Cardio"
        "Default_Level" "3"
        "Max_Unlockable_Level" "3"
        }
        {
        "Id" "Exercise"
        "Default_Level" "3"
        "Max_Unlockable_Level" "3"
        }
        {
        "Id" "Diving"
        "Default_Level" "0"
        "Max_Unlockable_Level" "0"
        }
        {
        "Id" "Parkour"
        "Default_Level" "3"
        "Max_Unlockable_Level" "3"
        }
        {
        "Id" "Sneakybeaky"
        "Default_Level" "4"
        "Max_Unlockable_Level" "4"
        }
        {
        "Id" "Vitality"
        "Default_Level" "3"
        "Max_Unlockable_Level" "3"
        }
        {
        "Id" "Immunty"
        "Default_Level" "0"
        "Max_Unlockable_Level" "0"
        }
        {
        "Id" "Toughness"
        "Default_Level" "3"
        "Max_Unlockable_Level" "3"
        }
        {
        "Id" "Strenght"
        "Default_Level" "0"
        "Max_Unlockable_Level" "0"
        }
        {
        "Id" "Warmblooded"
        "Default_Level" "5"
        "Max_Unlockable_Level" "5"
        }
        {
        "Id" "Survival"
        "Default_Level" "0"
        "Max_Unlockable_Level" "0"
        }
        {
        "Id" "Healing"
        "Default_Level" "5"
        "Max_Unlockable_Level" "5"
        }
        {
        "Id" "Crafting"
        "Default_Level" "3"
        "Max_Unlockable_Level" "3"
        }
        {
        "Id" "Outdoors"
        "Default_Level" "5"
        "Max_Unlockable_Level" "5"
        }
        {
        "Id" "Cooking"
        "Default_Level" "3"
        "Max_Unlockable_Level" "3"
        }
        {
        "Id" "Fishing"
        "Default_Level" "5"
        "Max_Unlockable_Level" "5"
        }
        {
        "Id" "Agriculture"
        "Default_Level" "6"
        "Max_Unlockable_Level" "6"
        }
        {
        "Id" "Mechanic"
        "Default_Level" "5"
        "Max_Unlockable_Level" "5"
        }
        {
        "Id" "Engineer"
        "Default_Level" "0"
        "Max_Unlockable_Level" "0"
        }
    ]
}

**Reply by WhiteBlaaack** [reply]
That's the file

**Reply by WhiteBlaaack** [reply]
![image](https://user-images.githubusercontent.com/74914249/126517602-cbd73e3f-8ef6-4734-8f99-c439ed3ee22e.png)

**Reply by SDGNelson** [reply]
Oh sorry I missed earlier: Type should be SDG.Unturned.LevelAsset not SDG.Unturned.SkillsAsset.

**Reply by WhiteBlaaack** [reply]
So you are saying that I should put it in the level asset?

**Reply by WhiteBlaaack** [reply]
![image](https://user-images.githubusercontent.com/74914249/126565050-704dba3a-ce0c-4f5c-b718-a374f6614ef8.png)
Cuz i have this one too

**Reply by WhiteBlaaack** [reply]
Nvm, I got it
